src/ralph/backends/lrs/fs.py

```diff
@@ -0,0 +1,348 @@
+            return statement.get("actor", {}).get("mbox_sha1sum") == sha1sum
+
+        def match_related_sha1sum(statement: dict) -> bool:
+            """Return `True` if the statement has any agent matching `sha1sum`."""
+            for agent in FSLRSBackend._get_related_agents(statement):
+                if agent.get("mbox_sha1sum") == sha1sum:
+                    return True
+
+            statement_object = statement.get("object", {})
+            if statement_object.get("objectType") == "SubStatement":
+                return match_related_sha1sum(statement_object)
+            return False
+
+        if sha1sum:
+            filters.append(match_related_sha1sum if related else match_sha1sum)
+
+    @staticmethod
+    def _add_filter_by_openid(
+        filters: list, openid: Union[str, None], related: Union[bool, None]
+    ) -> None:
+        """Add the `match_openid` filter if `openid` is set."""
+
+        def match_openid(statement: dict) -> bool:
+            """Return `True` if the statement has the given `actor.openid`."""
+            return statement.get("actor", {}).get("openid") == openid
+
+        def match_related_openid(statement: dict) -> bool:
+            """Return `True` if the statement has any agent matching `openid`."""
+            for agent in FSLRSBackend._get_related_agents(statement):
+                if agent.get("openid") == openid:
+                    return True
+
+            statement_object = statement.get("object", {})
+            if statement_object.get("objectType") == "SubStatement":
+                return match_related_openid(statement_object)
+            return False
+
+        if openid:
+            filters.append(match_related_openid if related else match_openid)
+
+    @staticmethod
+    def _add_filter_by_account(
+        filters: list,
+        name: Union[str, None],
+        home_page: Union[str, None],
+        related: Union[bool, None],
+    ) -> None:
+        """Add the `match_account` filter if `name` or `home_page` is set."""
+
+        def match_account(statement: dict) -> bool:
+            """Return `True` if the statement has the given `actor.account`."""
+            account = statement.get("actor", {}).get("account", {})
+            return account.get("name") == name and account.get("homePage") == home_page
+
+        def match_related_account(statement: dict) -> bool:
+            """Return `True` if the statement has any agent matching the account."""
+            for agent in FSLRSBackend._get_related_agents(statement):
+                account = agent.get("account", {})
+                if account.get("name") == name and account.get("homePage") == home_page:
+                    return True
+
+            statement_object = statement.get("object", {})
+            if statement_object.get("objectType") == "SubStatement":
+                return match_related_account(statement_object)
+            return False
+
+        if name or home_page:
```

Comment:
Should both `name` **and** `homepage` be provided?

Reply:
Good point! Thanks) Yes, indeed, if one is present, the other should be too) Updated this)